lib/mqtt/mqtt_client.dart

```diff
@@ -0,0 +1,141 @@
+    var mqttClientId = dotenv.env["MQTT_PORT"];
+    var mqttPort = int.tryParse(dotenv.env["MQTT_PORT"] ?? "");
+    _client = MqttServerClient.withPort(mqttBasePath!, mqttClientId!, mqttPort!);
+  }
+  
+  /// Connects MQTT Client if not already connected.
+  Future<void> connect() async {
+    if (_client.connectionStatus!.state == MqttConnectionState.connected) {
+      logger.info("MQTT Client already connected");
+      return;
+    }
+    
+    _client.logging(on: true);
+    _client.onConnected = onConnected;
+    _client.onDisconnected = onDisconnected;
+    _client.onUnsubscribed = onUnsubscribed;
+    _client.onSubscribed = onSubscribed;
+    _client.onSubscribeFail = onSubscribeFail;
+
+    final connMessage = MqttConnectMessage()
+        .keepAliveFor(60)
+        .withWillTopic(statusTopic)
+        .withWillMessage(StatusMessage(false).toJson().toString())
+        .startClean()
+        .withWillQos(MqttQos.atLeastOnce);
+    _client.connectionMessage = connMessage;
+    try {
+      await _client.connect();
+    } catch (e) {
+      logger.shout("Exception: $e");
+      _client.disconnect();
+    }
+
+    _client.updates?.listen((List<MqttReceivedMessage<MqttMessage>> c) {
+      final MqttPublishMessage message = c[0].payload as MqttPublishMessage;
+      final payload =
+      MqttPublishPayload.bytesToStringAsString(message.payload.message);
+
+      logger.info("Received message:$payload from topic: ${c[0].topic}>");
+    });
+  }
+  
+  /// Handler for successful connections event.
+  void onConnected() {
+    logger.info("Connected, sending status message...");
+    publishMessage(statusTopic, createMessagePayload(StatusMessage(true).toJson().toString()));
+  }
+
+  /// Handler for disconnection event.
+  void onDisconnected() {
+    logger.info("Disconnected");
+  }
+
+  /// Handler for subscribed to topic events.
+  void onSubscribed(String topic) {
+    logger.info("Subscribed to topic: $topic");
+  }
+
+  /// Handler for failing to subscribe to topic events.
+  void onSubscribeFail(String topic) {
+    logger.info("Failed to subscribe to: $topic");
+  }
+
+  /// Handler for unsubscribing from topic events.
+  void onUnsubscribed(String? topic) {
+    logger.info("Unsubscribed from topic: $topic");
+  }
+  
+  /// Disconnects MQTT Client
+  void disconnect() {
+    logger.info("Disconnecting...");
+    _client.disconnect();
+  }
+  
+  /// Reconnects MQTT Client.
+  /// 
+  /// In case of failing to connect, retries for 3 times.
+  Future<void> _reconnect({ retryAttempts = 3 }) async {
+    logger.info("Attempting to reconnect MQTT Client ($retryAttempts)...");
+    try {
+      await _client.connect();
+    } catch (e) {
+      if (retryAttempts > 0) {
+        _reconnect(retryAttempts: retryAttempts - 1);
+      } else {
+        throw Exception("Couldn't reconnect MQTT Client.");
+      }
+    }
+  }
+  
+  /// Gets MQTT Client connection status string.
+  String getClientConnectionStatus() {
```

Comment:
Could this be private, e.g. starting with _? It's atm used only from inside the class.

Reply:
Resolved.